R/write_po_file.R

```diff
@@ -300,7 +300,7 @@ format.po_metadata = function(x, template = FALSE, use_plurals = FALSE, ...) {
     x$email = "EMAIL@ADDRESS"
     x$language = ''
     x$language_team = "LANGUAGE <LL@li.org>"
-    x$charset = 'CHARSET'
+    x$charset = 'UTF-8'
```

Comment:
I'm not sure why this was CHARSET before, but I was getting an error from `tools::checkPoFile()` that it was failing to `iconv` `CHARSET` to `UTF-8`.

Reply:
Hmm note that .pot is not mentioned in `?tools::checkPoFile`. `CHARSET` shows up in templates in base:

https://github.com/wch/r-source/blob/bf5885960808d7c539e74b238575f54de0faba04/src/library/base/po/R.pot#L17

https://github.com/wch/r-source/blob/bf5885960808d7c539e74b238575f54de0faba04/src/library/base/po/R-base.pot#L10

Here's what I'm seeing in the `xgettext` docs:

https://www.gnu.org/software/gettext/manual/gettext.html

> `MIME-Version, Content-Type, Content-Transfer-Encoding`
> These values are set according to the content of the POT file and the current locale. If the POT file contains charset=UTF-8, it means that the POT file contains non-ASCII characters, and we keep the UTF-8 encoding. Otherwise, when the POT file is plain ASCII, we use the locale’s encoding.

and 

> `Content-Type`
> Replace ‘CHARSET’ with the character encoding used for your language, in your locale, or UTF-8. This field is needed for correct operation of the msgmerge and msgfmt programs, as well as for users whose locale’s character encoding differs from yours (see Charset conversion).

In the gettext sources, here is where `xgettext` initially populates the default `CHARSET`:

https://github.com/autotools-mirror/gettext/blob/030c0341a4a0ba6ad7fe62e83ff663bdc76cbe4d/gettext-tools/src/xgettext.c#L2015-L2028

Here `write-po` treats `CHARSET` as ASCII:

https://github.com/autotools-mirror/gettext/blob/030c0341a4a0ba6ad7fe62e83ff663bdc76cbe4d/gettext-tools/src/write-po.c#L1650-L1653

This snippet makes me thing setting UTF-8 is probably OK, since the default `CHARSET` has the probably-in-general faulty assumption that the strings in the .pot file are all-ASCII:

https://github.com/autotools-mirror/gettext/blob/030c0341a4a0ba6ad7fe62e83ff663bdc76cbe4d/gettext-tools/src/po-charset.c#L490-L509

That said, current r-devel .pot are all-ASCII:

```
grep -rP "[\x80-\xFF]" ~/svn/R-devel/ --include=*.pot
# <empty>
```

CRAN would require this to hold true for all packages at the R level, and probably the C level?

Reply:
We could probably patch `tools::checkPoTools` to accept the `CHARSET` :arrow_right: `ASCII` logic, but I'm not sure that would solve all of the issues of using `checkPoTools()` on .pot files when it was designed for .po.

Reply:
Maybe I'll back this change out here and we can discuss more thoroughly in a separate issue. I think potools should assume all `.pot` and `.po` files are UTF-8 and then fix any downstream problems that causes.

Reply:
Hmmm, backing out this change doesn't break any of my tests (or interactive usage) so definitely doesn't need to be here.